AnkiDroid/src/main/java/com/ichi2/anki/NavigationDrawerActivity.java

```diff
@@ -418,6 +418,18 @@ protected void openCardBrowser() {
         if (currentCardId != null) {
             intent.putExtra("currentCard", currentCardId);
         }
+        intent.putExtra("called_from_js_api", false);
+        startActivityForResultWithAnimation(intent, REQUEST_BROWSE_CARDS, START);
+    }
+
+    protected void openCardBrowser(String query) {
+        Intent intent = new Intent(NavigationDrawerActivity.this, CardBrowser.class);
+        Long currentCardId = getCurrentCardId();
+        if (currentCardId != null) {
+            intent.putExtra("currentCard", currentCardId);
+        }
+        intent.putExtra("called_from_js_api", true);
+        intent.putExtra("search_query", query);
```

Comment:
Change the official "openCardBrowser" API to add the query parameter as a `@Nullable` item and update all callers to the new API

Do this as a separate clean commit, then review of that change is trivial

As a second commit then add the Javascript API part that uses it - then this looks good to go?

```suggestion
    protected void openCardBrowser(String query) {
        Intent intent = new Intent(NavigationDrawerActivity.this, CardBrowser.class);
        Long currentCardId = getCurrentCardId();
        if (currentCardId != null) {
            intent.putExtra("currentCard", currentCardId);
        }
        intent.putExtra("search_query", query);
```